jwql/instrument_monitors/miri_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,566 @@
+            if 'uncal' in file_name:
+
+                # Define some useful parameters
+                dir_name = '_'.join(file_name.split('_')[:4])  # aka file_name[51:76]
+                self.nints = fits.getheader(file_name)['NINTS']
+
+                # Ensure the directory to the data exists.  If so, copy over files
+                observation_dir = os.path.join(self.data_dir, dir_name)
+                ensure_dir_exists(observation_dir)
+                try:
+                    copy_files([file_name], observation_dir)
+                except:
+                    logging.info('Failed to copy {} to observation dir.'.format(file_name))
+
+                # Next we run the pipeline on the files to get the proper outputs
+                uncal_file = os.path.join(observation_dir, os.path.basename(file_name))
+                try:
+                    pipeline_tools.calwebb_detector1_save_jump(uncal_file, observation_dir, ramp_fit=True, save_fitopt=False)
+                except:
+                    logging.info('Failed to complete pipeline steps on {}.'.format(uncal_file))
+
+                # Next we analyze the cosmic rays in the new data
+                for output_file in os.listdir(observation_dir):
+
+                    logging.info('output file')
+                    logging.info(output_file)
+
+                    if 'jump' in output_file:
+                        jump_filename = os.path.join(observation_dir, output_file)
+                    else:
+                        jump_filename = None
+
+                    if self.nints == 1:
+                        if '0_ramp_fit' in output_file:
+                            rate_file = os.path.join(observation_dir, output_file)
+                        else:
+                            rate_file = None
+
+                    elif self.nints > 1:
+                        if '1_ramp_fit' in output_file:
+                            rate_file = os.path.join(observation_dir, output_file)
+                        else:
+                            rate_file = None
+
+                try:
+                    jump_head, jump_data, jump_dq = self.get_jump_data(jump_filename)
+                except:
+                    logging.info('Could not open jump file: {}'.format(jump_filename))
+
+                try:
+                    rate_data = self.get_rate_data(rate_file)
+                except:
+                    logging.info('Could not open rate file: {}'.format(rate_file))
+
+                jump_locations = self.get_jump_locations(jump_dq)
+                jump_locations_pre = self.group_before(jump_locations)
+
+                eff_time = jump_head['EFFEXPTM']
+
+                # Get observation time info
+                obs_start_time = jump_head['EXPSTART']
+                obs_end_time = jump_head['EXPEND']
+                start_time = julian.from_jd(obs_start_time, fmt='mjd')
+                end_time = julian.from_jd(obs_end_time, fmt='mjd')
```

Comment:
@mengesser It seems like there is some logic that breaks down here.  I am seeing a lot of errors when testing things out in this part of the code.  Namely, the `pipeline_tools.calwebb_detector1_save_jump` step does not seem to complete sucessfully, which means that no `ramp_fit` or `jump` files are produced, and so the later parts of the code cannot be completed (they crash since variables like 'jump_head' are referenced before assignment).

Could you re-work this part of the algorithm to ensure that parts of the code either crash gracefully or get skipped appropriately?